### ISSUE TITLE
Get reports with actual conflation distance and deviation from merger analyzers

### ISSUE THREAD

**Issue by flacombe** [osm-fr/osmose-backend#2000]
Quality isn't always on/off about feature existence in OSM in regard of an external opendata file.
It can be measured according to conflation distance when comparing features against opendata.
The nearer osm and opendata match together, the better the quality actually is.

It would be nice to complete the current reporting with the average conflation distance and its deviation found during analysis, when applicable.

This would give a supplementary information about osm distance from opendata, particularly when conflation distance is high.
As an example, an analysis configured with 200m as conflation distance will show better results with an average measured distance of 20m instead of 100m inside the 200m circle.

It would show quality improvements over time when users spend time to improve feature position in OSM even if there is no warning to solve (when the feature already lies in the conflation boundary).

I don't really know if it is feasible in Osmose and how (probably with an additional SQL query involving ST_Distance).
Let's discuss it.

**Comment by phd0** [comment]
yes, that bothers me too. knowing that an average will be useless. 
A standard deviation would already be better, the dream would be a curve type curve of gausse.
In the sequel, I would almost have liked the distance to be a modifiable parameter for each user in Osmose, which would make it possible to first rough out the points that seem completely inconsistent, then to tackle the small adjustments. 
With the data has more than 6000 issues, it helps to be less discouraged from the start.

But who can analyze the python script to confirm that the tolerance distance is indeed a circle, and not a square?

**Comment by frodrigo** [comment]
Not sure about what do you mean both.

Once the conflation is made with a given distance. On the result only compute the mean, median distance and show it in all integration popup ?

It could not be done before conflation was done.

I confirm the distance is made using a circle.

**Comment by flacombe** [comment]
> I would almost have liked the distance to be a modifiable parameter for each user in Osmose

Adjusting conflation distance for every user would be hard: osmose would have to run every analysis for every user with a different conflation distance.
A more affordable change could be to filter warnings in the frontend with a user-defined limit took in the [0; analysis conflation distance[ range as to hide warnings about very small conflation distances and focus with big gaps.

It won't solve the power towers case as we shouldn't raise the conflation distance of the analysis, so the use won't be able to find only big gaps as the gap is small from the analysis perspective.
It's another topic, let's focus on conflation stats.

> Once the conflation is made with a given distance. On the result only compute the mean, median distance and show it in all integration popup ?

Yes, and display it as supplementary graphs like https://osmose.openstreetmap.fr/fr/issues/graph.png?item=8290&start_date=2023-01 for every analysis that could handle that.
It's not necessary in integration popups.

It only regards osm features that actually conflate with opendata and don't show missing warnings.

Example :
It should lead to avg=5.3, deviation=0.03
![conflation_stats](https://github.com/osm-fr/osmose-backend/assets/5690599/7dc69852-8376-43ef-aaef-fcbe54912a84)

**Comment by phd0** [comment]
> Adjusting conflation distance for every user would be hard: osmose would have to run every analysis for every user with a different conflation distance. A more affordable change could be to filter warnings in the frontend with a user-defined limit took in the [0; analysis conflation distance[ range as to hide warnings about very small conflation distances and focus with big gaps.
yes this idea like me !
I would even suggest adding in the table https://osmose.openstreetmap.fr/fr/issues/open?item=8290 (for example) a new column which indicates the distance with the nearest searched object (that it is conflicted or not). So we could sort in descending order, or have an idea of the distance with the near object.
I am not an expert in osmosis scripts, and I do not realize the work required, nor its feasibility.
